File: daily_challenge_leetcode/2257.js
var countUnguarded = function (m, n, guards, walls) {
  const mark_guarded = (row, col, grid) => {
    for (let i = row - 1; i >= 0; i--) {
      if (grid[i][col] === 2 || grid[i][col] === 3) {
        break;
      }
      grid[i][col] = 1;
    }
    for (let i = row + 1; i < grid.length; i++) {
      if (grid[i][col] === 2 || grid[i][col] === 3) {
        break;
      }
      grid[i][col] = 1;
    }
    for (let j = col - 1; j >= 0; j--) {
      if (grid[row][j] === 2 || grid[row][j] === 3) {
        break;
      }
      grid[row][j] = 1;
    }
    for (let j = col + 1; j < grid[0].length; j++) {
      if (grid[row][j] === 2 || grid[row][j] === 3) {
        break;
      }
      grid[row][j] = 1;
    }
  };
  const grid = Array.from({ length: m }, () => Array(n).fill(0));
  for (const [i, j] of guards) {
    grid[i][j] = 2;
  }
  for (const [i, j] of walls) {
    grid[i][j] = 3;
  }
  for (const [i, j] of guards) {
    mark_guarded(i, j, grid);
  }
  let count = 0;
  for (let i = 0; i < m; i++) {
    for (let j = 0; j < n; j++) {
      if (grid[i][j] === 0) {
        count++;
      }
    }
  }
  return count;
};

File: daily_challenge_leetcode/2257.py
from typing import List

class Solution:
    def mark_guarded(self, row: int, col: int, grid: List[List[int]]) -> None:
        for i in range(row - 1, -1, -1):
            if grid[i][col] == 2 or grid[i][col] == 3:
                break
            grid[i][col] = 1
        for i in range(row + 1, len(grid)):
            if grid[i][col] == 2 or grid[i][col] == 3:
                break
            grid[i][col] = 1
        for j in range(col - 1, -1, -1):
            if grid[row][j] == 2 or grid[row][j] == 3:
                break
            grid[row][j] = 1
        for j in range(col + 1, len(grid[0])):
            if grid[row][j] == 2 or grid[row][j] == 3:
                break
            grid[row][j] = 1

    def countUnguarded(
        self, m: int, n: int, guards: List[List[int]], walls: List[List[int]]
    ) -> int:
        grid = [[0] * n for _ in range(m)]
        for i, j in guards:
            grid[i][j] = 2
        for i, j in walls:
            grid[i][j] = 3
        for i, j in guards:
            self.mark_guarded(i, j, grid)
        count = 0
        for i in range(m):
            for j in range(n):
                if grid[i][j] == 0:
                    count += 1

        return count
